mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SearchUtil.java

```diff
@@ -171,6 +173,18 @@ public static String getMailboxAddress(String headerValue) {
         }
         return "";
     }
+
+
+    public static String getSerializedMessageIdIfSupportedByUnderlyingStorageOrNull(MailboxMessage mailboxMessage) {
+        try {
+            MessageId messageId = mailboxMessage.getMessageId();
+            if (messageId != null) {
+                return messageId.serialize();
+            }
```

Comment:
I think we can't use Optional here because of Java 6. I don't assume anything about getMessage because it's comportment when the backend is unsupported is not defined I considere all possible case. We could choose a default comportment for backend that does not implement it but I think it will be out of the scope of this PR.